src/output-json-mqtt.c

```diff
@@ -137,6 +141,7 @@ static OutputInitResult OutputMQTTLogInitSub(ConfNode *conf,
     JsonMQTTLogParseConfig(conf, mqttlog_ctx);
 
     AppLayerParserRegisterLogger(IPPROTO_TCP, ALPROTO_MQTT);
+    ajt->applayer_ctx[ALPROTO_MQTT] = mqttlog_ctx;
```

Comment:
How can we do that in rust ?

Now `AppLayerParserRegisterLogger` is supposed to be called from parser registration

And `OutputJsonLogInitSub` is supposed to be used by all